api/v1alpha4/cluster_types.go

```diff
@@ -60,6 +60,69 @@ type ClusterSpec struct {
 	// for provisioning infrastructure for a cluster in said provider.
 	// +optional
 	InfrastructureRef *corev1.ObjectReference `json:"infrastructureRef,omitempty"`
+
+	// This encapsulates the topology for the cluster.
+	// NOTE: This feature is alpha; it is required to enable the ClusterManagedTopologies
+	// feature gate flag to activate managed topologies support.
+	// +optional
+	Topology *Topology `json:"topology,omitempty"`
+}
+
+// Topology encapsulates the information of the managed resources.
+type Topology struct {
+	// The name of the ClusterClass object to create the topology.
+	Class string `json:"class,omitempty"`
```

Comment:
omitempty not needed here?

Reply:
should this be a `corev1.LocalObjectReference`?

Reply:
Classes are always by name and enforced to be in the same Namespace